### PR TITLE
fix(auth): add auth header in provider service

### PR DESCRIPTION
**Pull request by rohitkrai03**
Fixes https://github.com/openshiftio/openshift.io/issues/4268
Fixes https://github.com/fabric8-services/fabric8-auth/issues/633

### CODE REVIEW

**Comment by alien-ike** [comment]
### Ike Plugins (test-keeper)

Thank you @rohitkrai03 for this contribution!



It appears that no tests have been added or updated in this PR.

Automated tests give us confidence in shipping reliable software. Please add some as part of this change.

If you are an admin or the reviewer of this PR and you are sure that no test is needed then you can use the command `/ok-without-tests` as a comment to make the status green.


For more information please head over to official [documentation](http://arquillian.org/ike-prow-plugins/#_test_keeper_plugin). You can find there how to configure the plugin.

**Comment by rohitkrai03** [comment]
@jiekang Can you review this? Its a SEV-1 issue.

**Comment by sbose78** [comment]
Hi @rohitkrai03 , Would it be reasonable to update the test to try validating the presence of the header to https://github.com/fabric8-ui/fabric8-ui/blob/master/src/app/shared/account/provider.service.spec.ts ?

**Comment by rohitkrai03** [comment]
@sbose78 The tests actually validate presence of auth header by mocking authService https://github.com/fabric8-ui/fabric8-ui/blob/5ea0a594b37587905c96c184e647ec3945c4b1ab/src/app/shared/account/provider.service.spec.ts#L28. that is the reason this was not caught since all tests were passing.

**Comment by rohitkrai03** [comment]
@sbose78 Actually no, the token was for validating something else. I am adding tests to verify auth header.

**Comment by rohitkrai03** [comment]
@jiekang I've added tests for header. Also, https://github.com/fabric8-ui/fabric8-ui/pull/3285 is for replicating `Fabric8UIHttpService` by use of interceptors. I am working on adding auth interceptor from `ngx-login-client` which adds header for `HttpClient` request.

File: src/app/shared/account/provider.service.spec.ts [hunk 1]
```diff
@@ -12,11 +12,12 @@ import { ProviderService } from './provider.service';
 
 import { createMock } from 'testing/mock';
 
-describe('Service: Provider Service', () => {
+fdescribe('Service: Provider Service', () => {
```
Comment (jiekang) [inline src/app/shared/account/provider.service.spec.ts]: I think this is for local testing?

Comment (rohitkrai03) [reply src/app/shared/account/provider.service.spec.ts]: Sorry, removed.

File: src/app/shared/account/provider.service.spec.ts [hunk 2]
```diff
@@ -60,6 +60,7 @@ describe('Service: Provider Service', () => {
 
       const req: TestRequest = controller.expectOne('http://example.com/api/token?force_pull=true&for=https://github.com');
       expect(req.request.method).toEqual('GET');
+      expect(req.request.headers.get('Authorization')).toEqual(`Bearer ${mockToken}`);
```
Comment (sbose78) [inline src/app/shared/account/provider.service.spec.ts]: Thanks!

**Comment by joshuawilson** [comment]
We should consider removing the customer http service with the addition of the new httpclient.

**Pull request merged**